source/_static/js/main.js

```diff
@@ -504,15 +504,36 @@ window.addEventListener("DOMContentLoaded", (event) => {
       });
     }
   })();
+
 // ------------------------------------------------
 // Tab switching for hidden anchors
 // ------------------------------------------------
 
-    document.getElementById((document.getElementById(window.location.hash.substring(1)))
-        .closest("div")
-        .previousSibling.getAttributes("for"))
-    .click();
+  // Target ID - where the browser should scroll to
+  const hashElem = document.getElementById(window.location.hash.substring(1));
+
+  // If the element is hidden, neither scrollIntoView nor hashScroll will work.
+  // This function checks whether the element is visible or not.
+  function isHashElemeVisible(elem) {
+    var style = window.getComputedStyle(elem);
+    return (elem.offsetWidth > 0 || elem.offsetHeight > 0) && style.display !== 'none' && style.visibility !== 'hidden';
+  }
+
+  // Check if the hash is valid
+  // If it is, but not visible, the corresponding tab is also not visible.
+  // Make the tab active and scroll to it
+  if(hashElem) {
+    if(!isHashElemeVisible(hashElem)) {
+      hashElem.closest(".sd-tab-content").previousElementSibling.click();
 
+      setTimeout(() => {
+        hashElem.scrollIntoView();
+      }, 100);
+    }
+    else {
+      hashElem.scrollIntoView();
+    }
+  }
 });
 
 
```

Comment:
@rushenn do we duplicate the code from above down here, or is this chunk OK as is?

Reply:
@ravindk89 

I've added the `else` part because in some cases, I've noticed that Chrome fails to scroll even when the hash is valid. So, adding `scrollIntoView` ensures consistent scrolling across all browsers.

The code below is no longer needed. Can we remove it, @feorlen?

```
window.addEventListener("hashchange", function() {
  document.getElementById((document.getElementById(window.location.hash.substring(1)))
          .closest("div")
	  .previousSibling.getAttribute("for"))
        .click();
});
```

Reply:
Also, we might need to disable the smooth scroll behavior on `_layout.scss@51` and `_base.scss@14`. Smooth scrolling might prevent the browser from completing the scroll. We can test this behavior once merged and make this change only if necessary.

Reply:
@rushenn  Dead code removed and PR restaged. It's not working for me, however.
http://192.241.195.202:9000/staging/link-into-tab/linux/reference/minio-server/settings/console.html#content-security-policy

Reply:
It seems to be working for me. I tried the following URL, and I can see that the tab is opened and scrolled to the given anchor as well.

Do you see any errors related to the function in the browser console? 

http://192.241.195.202:9000/staging/link-into-tab/linux/reference/minio-server/settings/console.html#mc-conf.browser.hsts_seconds